src/libostree/ostree-repo-commit.c

```diff
@@ -1738,26 +1791,31 @@ ostree_repo_prepare_transaction (OstreeRepo     *self,
     return FALSE;
 
   self->in_transaction = TRUE;
-  if (self->min_free_space_percent > 0)
+  self->cleanup_stagedir = FALSE;
+  if (self->min_free_space_percent >= 0 || self->min_free_space_mb >= 0)
```

Comment:
`ostree_repo_prepare_transaction` is a special case: 
We need `>=` as that will lead to calculate the `reserved_blocks`. Therefore, if any one is mentioned to be `0` this will still be`TRUE`,  calculate `reserved_blocks`=0 and finally set `self->txn.max_block` (see below) accordingly.

And if you notice `self->txn.max_block` is how we decide we are over the  free-space quota or not. Once it is computed (in `prepare_transaction`) we can happily use `>` for `min-free-space-*` (which is what is happening). Does it make sense?

Reply:
If you always need to take that branch, why have the `if` statement?

Reply:
Point.